platform-sdk/swirlds-merkledb/src/timingSensitive/java/com/swirlds/merkledb/MerkleDbSnapshotTest.java

```diff
@@ -110,13 +109,21 @@ void setupTest() throws Exception {
     }
 
     @AfterEach
-    public void afterTest() {
+    public void afterTest() throws IOException {
         // check db count
         AssertionUtils.assertEventuallyEquals(
                 0L,
                 MerkleDbDataSource::getCountOfOpenDatabases,
                 Duration.ofSeconds(1),
                 "Expected no open dbs. Actual number of open dbs: " + MerkleDbDataSource.getCountOfOpenDatabases());
+
+        if (Files.exists(tempDirectory)) {
+            Files.walk(tempDirectory).map(Path::toFile).forEach(file -> {
+                if (!file.delete()) {
+                    file.deleteOnExit();
+                }
+            });
+        }
```

Comment:
Are there files created by this test that don't get deleted?

Reply:
This was is an attempt get rid of the `CleanupMode.NEVER` similar to the one mentioned in the [above comment](https://github.com/hashgraph/hedera-services/pull/14695#discussion_r1781565596). Still in progress though.